### ISSUE TITLE
Display matching invocations alongside their respective expectations in error message

### ISSUE THREAD

**Issue by floehopper** [freerange/mocha#178]
See #167. Display all the actual invocations recorded against both unsatisfied & satisfied expectations, including the parameter values passed into the method and the return value generated by the expectation. I'm thinking along the following lines:

``` ruby
foo = mock('foo')
foo.expects(:bar).with(1).returns('a')
foo.stubs(:bar).with(2).returns('b').then.returns('c').then.raises(StandardError)

3.times { foo.bar(2) }
foo.bar(3)
```

```
unexpected invocation: #<Mock:foo>.bar(3)
unsatisfied expectations:
- expected exactly once, not yet invoked: #<Mock:foo>.bar(1)
satisfied expectations:
- allowed any number of times, invoked 3 times: #<Mock:foo>.bar(2)
  - #<Mock:foo>.bar(2) # => 'b'
  - #<Mock:foo>.bar(2) # => 'c'
  - #<Mock:foo>.bar(2) # => raised StandardError
```

It would also be nice to record any yields that occur.

We might only want to enable this using a verbose configuration option.

**Comment by floehopper** [comment]
Implemented in #394 & #410.

**Comment by floehopper** [comment]
See also https://github.com/freerange/mocha/issues/632.
